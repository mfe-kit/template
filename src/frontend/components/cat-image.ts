import { namespace } from '../config';

export const CatImage = (imgUrl: string, isLoading: boolean): string => {
  return `
    <div class="${namespace}-img ${isLoading ? namespace + '-hide' : ''}">
      <img class="${namespace}-image" src="${imgUrl}" alt="cat" />
    </div>
  `;
};
