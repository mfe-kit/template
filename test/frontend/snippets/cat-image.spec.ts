import { describe, it, expect } from 'vitest';
import { CatImage } from '../../../src/frontend/snippets/cat-image';

describe('CatImage', () => {
  it('should show image', async () => {
    const result = CatImage('testUrl', false);
    expect(result).toMatchSnapshot();
  });
  it('should hide image', async () => {
    const result = CatImage('testUrl', true);
    expect(result).toMatchSnapshot();
  });
});
