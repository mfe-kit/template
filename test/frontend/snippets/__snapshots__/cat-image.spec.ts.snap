// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CatImage > should hide image 1`] = `
"
    <div class="mfe-kit-template-img mfe-kit-template-hide">
      <img class="mfe-kit-template-image" src="testUrl" alt="cat" />
    </div>
  "
`;

exports[`CatImage > should show image 1`] = `
"
    <div class="mfe-kit-template-img ">
      <img class="mfe-kit-template-image" src="testUrl" alt="cat" />
    </div>
  "
`;
